File: 1079-sum-of-root-to-leaf-binary-numbers/sum-of-root-to-leaf-binary-numbers.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function sumRootToLeaf(root: TreeNode | null): number {
    let dfs = (curr:TreeNode,val:String)=>{
        if (curr === null){
            return 0;
        }
        if (curr.left === null && curr.right === null){
            const bin = String(val)+curr.val
            let ret = parseInt(bin,2)
            return ret
        }
        return dfs(curr.left,String(val)+curr.val) +dfs(curr.right,String(val)+curr.val)
    }
    return dfs(root,"")
};

File: 1079-sum-of-root-to-leaf-binary-numbers/sum-of-root-to-leaf-binary-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        
        def dfs(curr,val):
            if not curr:
                return 0
            if not curr.left and not curr.right:
                return int(val+str(curr.val),2)
            return dfs(curr.left,val+str(curr.val))+dfs(curr.right,val+str(curr.val))
        return dfs(root,"")